### DISCUSSION TITLE
LibreElec Matrix Beta27 Delay between playlist tracks

### DISCUSSION THREAD

**Discussion by AndrewNNZ** [Goldenfreddy0703/plugin.audio.ytmusic.exp#37] [Q&A]
Hi

Thanks for a great addon. I am running it on LibreElec with Matrix on a Raspberry Pi. It is working pretty well but I have a few questions:

With Beta 21 it changed tracks quickly but with Beta 27 it is very slow between tracks. Also very slow to start playing on first selected track too. Are you aware of this?

The other thing is that after a while it appears to crash and LibreElec reboots.

Thanks for all your hard work

AndrewNNZ

**Comment by Goldenfreddy0703** [comment]
Hey there, I'm not really aware of this but I think reason why it maybe loading slower is probably due to whatever happened to pytube a few releases ago. @fischcode may know the answer so hopefully he will respond soon in this.

**Reply by AndrewNNZ** [reply]
Ok, thanks

**Reply by AndrewNNZ** [reply]
I rolled back to Beta 25 and it is much quicker to start the first track and changes tracks instantly. So yeah looking at the change logs its probably the change from pytube to pytubefix as you suggested or possibly the change in the YT api.

**Comment by fischcode** [comment]
I can confirm this observation: the latest release of pytubefix uses what appears to be a JavaScript interpreter written in Python to extract the necessary information from Google‘s obfuscated code whereas previous versions used a much simpler approach making assumptions about the function to find. One could try to continue and use the simpler approach as long as it still works…What do you think?Am 15.05.2024 um 11:15 schrieb AndrewNNZ ***@***.***>:﻿
I rolled back to Beta 25 and it is much quicker to start the first track and changes tracks instantly. So yeah looking at the change logs its probably the change from pytube to pytubefix as you suggested or possibly the change in the YT api.

**Comment by AndrewNNZ** [comment]
I reckon revert the change to using pytubefix and see how it goes. Is pytubefix needed for the latest YT api or will it work with pytube? If you make another build I can check it. Hopefully it is the change to pytube and not something else.

**Comment by fischcode** [comment]
Pytube hasn’t been updated for quite a while and is no longer an option. Instead, we have to look at different versions of pytubefix or actually a combination of them. I will try to put something together for testing during the next days…Am 15.05.2024 um 21:40 schrieb AndrewNNZ ***@***.***>:﻿
I reckon revert the change to using pytubefix and see how it goes. Is pytubefix needed for the latest YT api or will it work with pytube? If you make another build I can check it. Hopefully it is the change to pytube and not something else.

**Reply by AndrewNNZ** [reply]
Ok, sounds good - thanks, let us know when you have a new version

**Reply by fischcode** [reply]
I am very sorry, but it turns out that reverting to the cipher.py from pytube 4.0 breaks playback of channel episodes. So, if fast random switching between songs is important to you, I suggest that you stick to beta 26 for as long as it still plays the songs you want to listen to. Eventually, only the js interpreter used by pytubefix 5 will work consistently and irrespective of changes to the obfuscation scheme ...

**Comment by AndrewNNZ** [comment]
Ok, no problem -thanks again for addon
